test/jdk/javax/swing/JButton/HtmlButtonWithTextAndIcon.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+/* @test
+ * @bug 8302173
+ * @requires (os.family == "mac")
+ * @summary Tests Aqua button icon positioning with HTML text
+ * @run main HtmlButtonWithTextAndIcon
+ */
+
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import javax.imageio.ImageIO;
+import javax.swing.Icon;
+import javax.swing.JButton;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB;
+
+public class HtmlButtonWithTextAndIcon {
```

Comment:
Would it be a good idea to combine `HtmlButtonWithTextAndIcon`  & `HtmlButtonImageTest` into one comprehensive test, since they are similar? This would also make sure that all the cases related to JButton's HTML ImageView are available at one place.

Reply:
I think it's OK from my perspective to have separate tests here because one relates to an icon while the other handles HTML images.